src/driver/mysql/MysqlDriver.ts

```diff
@@ -231,7 +231,7 @@ export class MysqlDriver implements Driver {
      */
     prepareHydratedValue(value: any, columnMetadata: ColumnMetadata): any {
         if (columnMetadata.type === Boolean) {
-            return value ? true : false;
```

Comment:
some driver (like mysql) does not have boolean type and return 1 or 0 so we have to cast it to boolean. Thats why your tests are failing